lib/src/signed_video_h26x_auth.c

```diff
@@ -128,9 +146,31 @@ verify_gop_hash(signed_video_t *self)
   return (memcmp(gop_info->computed_gop_hash, self->received_gop_hash, hash_size) == 0);
 }
 
-/**
- * Adds the hashes of NALUs used in computing the GOP hash and marks these NALUs as part of the GOP
- * hash.
+/*
+ * Iterates through the NALU list to find the first NALU used in the GOP hash.
+ * If the linked hash has not yet been updated with this NALU's hash, it updates the
+ * linked hash with the first NALU hash and marks it as used.
+ */
+void
+update_link_hash_for_auth(signed_video_t *self, h26x_nalu_list_t *nalu_list)
+{
+  const size_t hash_size = self->verify_data->hash_size;
+  h26x_nalu_list_item_t *item = nalu_list->first_item;
+  bool found_linked_hash = false;
+  while (item && !found_linked_hash) {
+    if (item->nalu->is_last_nalu_part && item->used_in_gop_hash &&
```

Comment:
Since you rely on `used_in_gop_hash` to determine which item to use for updating the linked hash, it is vulnerable to order of operation and whether the gop hash has been computed or not. Can it happen that `used_in_gop_hash` is not set and items removed from the list before the first SEI arrives?

In addition, isn't `is_last_nalu_part` always `true`?